### ISSUE TITLE
macOS vcpkg install hpx fails when installing hwloc

### ISSUE THREAD

**Issue by dokipen3d** [microsoft/vcpkg#3307]
```
Package: hwloc:x64-osx
  Vcpkg version: 0.0.110-unknownhash
```
looking at the log it seems to be looking for a windows.h....

```
In file included from /Users/dokipen/Documents/Projects/vcpkg/buildtrees/hwloc/src/hwloc-hwloc-1.11.7/./include/hwloc.h:53:
include/hwloc/autogen/config.h:29:10: fatal error: 'windows.h' file not found
#include <windows.h>
         ^~~~~~~~~~~
```
could be a define somewhere that make the build system think its running in windows.

**Comment by zbahhah** [comment]
I face same problem in Linux ubuntu 16.04 LTS: 

terminal command: $ ./vcpkg install hwloc curl
=====================

terminal output:
=====================
-- Build x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_build_cmake.cmake:170 (message):
    Command failed: ~/vcpkg/downloads/tools/cmake-3.10.2-linux/cmake-3.10.2-Linux-x86_64/bin/cmake;--build;.;--config;Debug;--target;install;--;-v;-j1
    Working Directory: ~/vcpkg/buildtrees/hwloc/x64-linux-dbg
    See logs for more information:
      ~/vcpkg/buildtrees/hwloc/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/hwloc/portfile.cmake:20 (vcpkg_install_cmake)
  scripts/ports.cmake:72 (include)
=====================


log output:
=====================

Error: Building package hwloc:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: hwloc:x64-linux
  Vcpkg version: 0.0.110-unknownhash

Additionally, attach any relevant sections from the log files above.

[1/20] /usr/bin/cc -D_CRT_SECURE_NO_WARNINGS -I~/vcpkg/buildtrees/hwloc/src/loc-1.11.7-b9e831bc50/./include -I~/vcpkg/buildtrees/hwloc/src/loc-1.11.7-b9e831bc50/./src -Iinclude -g -MD -MT CMakeFiles/libhwloc.dir/src/topology-custom.c.o -MF CMakeFiles/libhwloc.dir/src/topology-custom.c.o.d -o CMakeFiles/libhwloc.dir/src/topology-custom.c.o   -c ~/vcpkg/buildtrees/hwloc/src/loc-1.11.7-b9e831bc50/src/topology-custom.c
FAILED: CMakeFiles/libhwloc.dir/src/topology-custom.c.o 
/usr/bin/cc -D_CRT_SECURE_NO_WARNINGS -I~/vcpkg/buildtrees/hwloc/src/loc-1.11.7-b9e831bc50/./include -I~/vcpkg/buildtrees/hwloc/src/loc-1.11.7-b9e831bc50/./src -Iinclude -g -MD -MT CMakeFiles/libhwloc.dir/src/topology-custom.c.o -MF CMakeFiles/libhwloc.dir/src/topology-custom.c.o.d -o CMakeFiles/libhwloc.dir/src/topology-custom.c.o   -c ~/vcpkg/buildtrees/hwloc/src/loc-1.11.7-b9e831bc50/src/topology-custom.c
In file included from ~/vcpkg/buildtrees/hwloc/src/loc-1.11.7-b9e831bc50/./include/hwloc.h:53:0,
                 from ~/vcpkg/buildtrees/hwloc/src/loc-1.11.7-b9e831bc50/src/topology-custom.c:7:
include/hwloc/autogen/config.h:29:21: fatal error: windows.h: No such file or directory
compilation terminated.

**Comment by JackBoosY** [comment]
Please track issue #10602.
